sympy/plotting/tests/test_plot.py

```diff
@@ -623,3 +635,30 @@ def test_issue_13516():
     p = plot(sin(x), show=False)
     assert p.backend == DefaultBackend
     assert len(p[0].get_segments()) >= 30
+
+def test_check_figures_equal():
+    from sympy.testing.tmpfiles import TmpFileManager
+    matplotlib = import_module('matplotlib' , min_module_version='1.1.0', catch= (RuntimeError,))
+    if not matplotlib:
+        skip("Matplotlib not the default backend")
+    from matplotlib.testing.compare import compare_images
+    test_directory = os.path.dirname(os.path.abspath(__file__))
+
+    def tmp_file(dir=None, name=''):
+        return NamedTemporaryFile(suffix='.png', dir=dir, delete=False).name
+    try:
+        temp_dir = mkdtemp()
+        TmpFileManager.tmp_folder(temp_dir)
+        x = Symbol('x')
+        test_filename = tmp_file(dir = temp_dir, name = "test_singularity1")
+        cmp_filename = os.path.join(test_directory, "test_singularity1.png")
+        p = plot(x/(x-1)*(x-2))
+        p.save(test_filename)
+        compare_images(cmp_filename , test_filename , 0.005)
+        test_filename = tmp_file(dir = temp_dir, name = "test_singularity2")
+        cmp_filename = os.path.join(test_directory, "test_singularity2.png")
+        p = plot(sin(x)/x)
+        p.save(test_filename)
+        compare_images(cmp_filename , test_filename , 0.005)
```

Comment:
I think we need to have a more general discussion about whether we want to do this sort of test in SymPy. For one thing, it requires checking in binary files to the repo, which can bloat the repo size. Another issue is that if matplotlib changes their plot format in some way, it would break the test. It would also happen if we change the way the plot is formatted in some way.  Is there some way we can test the plot without comparing binary files?

Reply:
I added this type of test so that the extra line which get into plot can be compared with the desired plot. But currently, it is failing to produce test error.

> For one thing, it requires checking in binary files to the repo, which can bloat the repo size. 

This problem can be handled by a suggestion from @sylee957 who suggested to store the testing file to somewhere like sympy wiki. 
>Is there some way we can test the plot without comparing binary files?

I think we can do this by calculating the RMS value directly from the points which are used to plot the graph. This will remove our dependency on matplotlib for image comparison.

Reply:
Can you query the line data from `p._backend.ax[0]` somehow?

Reply:
As far as I understand the question yes we can find the line data from both the backend.